auction-server/src/auction/service/verification.rs

```diff
@@ -673,11 +783,16 @@ impl Verification<Svm> for Service<Svm> {
             transaction:        bid.chain_data.transaction.clone(),
         };
         let permission_key = bid_chain_data.get_permission_key();
+        let submit_type = self.get_submission_state(&permission_key).await;
+        assert!(
+            bid_data.bid_type == BidType::SubmitBid && submit_type == entities::SubmitType::ByOther,
+            "Invalid bid format for swap, should not use submit_bid instruction"
+        );
         tracing::Span::current().record("permission_key", bid_data.permission_account.to_string());
         self.check_deadline(&permission_key, bid_data.deadline)
             .await?;
-        self.verify_signatures(&bid, &bid_chain_data).await?;
-        // TODO we should verify that the wallet bids also include another instruction to the swap program with the appropriate accounts and fields
+        self.verify_signatures(&bid, &bid_chain_data, submit_type)
```

Comment:
I suggested to pass submit_type to this function. But I think it's not necessary as the opportunity exists check, only query the in_memory opportunity and it's fast enough. So I think it's not important to pre compute this value.